src/libCli/Call.cpp

```diff
@@ -189,7 +189,19 @@ namespace cli
                 deadline = std::chrono::system_clock::now() + std::chrono::milliseconds(customTimeoutMs);
             }           
         }
-              
+
+        //before calling the RPC, close the DescDb connection with a timeout.
+        grpc::Status dbDescStatus = ConnectionManager::getInstance().closeDescDbWithDeadline(serverAddress, deadline);
```

Comment:
I was about to suggest moving all the closing into destructors and just drop the descriptor database / proxy when we do not need it any more.
However the `ConnectionManager` singleton makes life hard here. gWhisper does not have the most beautiful design :-( 
So I am fine with it as is. what do you think?

Reply:
yeah ConnectionManager actually makes it hard. since we are also using shared ptrs here we would know how many owners are there for the proxy and when will it actaully closes.
Doing explicitly allows use to check the grpc status/result from calling the stream finish() aswell.